.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Allow manual runs. It is necessary to run the workflow manually, when the "formatter" workflow applies any changes.
+  workflow_dispatch:
+
+jobs:
+  Processor_Partial:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest, macos-13 ] # macos-13 for x86_x64 arch
+        framework: [ net8.0 ]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup .NET SDK v8.0.x
+        if: matrix.framework == 'net8.0'
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+      - name: Build
+        run: dotnet build G-Research.OpenTelemetry.Processor.Partial --configuration=Release --framework=${{ matrix.framework }}
+      - name: Run tests
+        shell: bash
+        run: dotnet test G-Research.OpenTelemetry.Processor.Partial.Tests --configuration=Release --no-build -v=Normal --framework=${{ matrix.framework }}
```

Comment:
I would fold these into just one `dotnet test` command that runs for the _solution_.
Something like:
```sh
dotnet test --configuration=Release -v=Normal --framework=${{ matrix.framework }}
```

Reply:
done.